### ISSUE TITLE
NeoVim crashes with terminfo 1.616

### ISSUE THREAD

**Issue by lbschenkel** [neovim/neovim#7206]
- `nvim --version`: **0.2.0**
- Vim 7.4 behaves differently? **does not crash**
- Operating system/version: **Ubuntu 16.04 and macOS 10.12 (MacPorts)**
- Terminal name/version: **any**
- `$TERM`: **tmux** or **vte** or **xterm** (probably more)

### Steps to reproduce

`nvim -u NORC` does not crash

but when **both** the following lines are present in init.vim:
```
set colorcolumn=80
set termguicolors
```
and:
- terminfo 1.616 (from http://invisible-island.net/datafiles/current/terminfo.src.gz) is compiled with `tic -x` (without `-x` the crash does not happen)
- `$TERM` is set to `tmux`, `vte` or `xterm` (not exhaustive, there are probably others)

then `nvim` crashes:

```
nvim: unibilium.c:735: unibi_get_str: Assertion `v > unibi_string_begin_ && v < unibi_string_end_' failed.
zsh: abort (core dumped)  nvim
```
Since this could be a bug in `unibilium`, I have filed a bug there and included more detailed info:
https://github.com/mauke/unibilium/issues/24

I have bisected the changes in terminfo for every ncurses release and the change that triggers the crash was introduced in terminfo 1.616 (included in ncurses-6.0-20170819).

**Comment by fornwall** [comment]
Check this pull request which fixes the problem for me: https://github.com/neovim/neovim/pull/7204

**Comment by lbschenkel** [comment]
I tested the fix (da84f7dcf9dab28a03b9a776f04d882c36fd77fb) and it indeed does not crash. However, I also tested against the parent commit (e5565891af5858965a65366ecb7f791abc988931) and it also does not crash. I re-tested against 0.2.0 and it does crash.

Show I try bisecting?

**Comment by justinmk** [comment]
@lbschenkel sure, it could be valuable. We had a big TUI refactor, so things were shuffled around and I would guess the crash will happen just before https://github.com/neovim/neovim/commit/105d680aea9f . It's possible that the shuffling-around covered up the bug, but @fornwall's fix still makes sense to me.

**Comment by lbschenkel** [comment]
The commit that fixed the crash for me was 133ae5eeeff3b83e1a3e6da35b57dcdfb92287b0.
